### DISCUSSION TITLE
Explicit support for trio

### DISCUSSION THREAD

**Discussion by thearchitector** [Neoteroi/BlackSheep#405] [Ideas]
I think it would be neat for BlackSheep to explicitly support Trio as an underlying async engine. I say explicitly because, for the most part, it seems like it does (having played around running an app via hypercorn on trio-mode). There are just a few places, from a very brief search namely the http client and gzip middleware, where there are calls to the std asyncio module that fail; in the case of gzip, it's because `get_running_loop` has no running asyncio loop to return (the failure is what prompted this question).

Are there any ideological reasons against adding support through a common interface like anyio? The only one that I can think of atm is the bulk of the added dependency for a few exceptions, but I'm not sure how one would do it with an optional dependency (unless perhaps, a disclaimer saying "if you want to use gzip or the http client, you need to install anyio as well").

If not, I'd be happy to put some time into a PR because I think this is a really cool project with designs i really agree with.

**Comment by thearchitector** [comment]
i see jinja's async mode also assumes asyncio, so perhaps that would have to be another disclaimer

**Comment by RobertoPrevato** [comment]
Hi @thearchitector 
Thank You for your recommendations. There are no ideological reasons against adding support for `trio` from my side. The only reason I didn't try it is because I didn't have the time to become familiar with `trio`, so lack of time is the reason here.

I welcome a PR. 👍🏼

**Reply by thearchitector** [reply]
awesome! I'll see what I can do 😄
